Eliminate modelmapper add custom mappers

SergiGnz opened the pull request.

Commit:
[wip] crear dtos, custom mappers, eliminar modelmapper

Commit:
Ignore files inside .idea folder

Commit:
Change modelMapper by a customMapper

Commit:
Delete ConverterUtils

Commit:
Fix test

amarvila commented on project-x/src/main/java/com/everis/d4i/project_x/controller/rest/model/CustomerRest.java:
```diff
@@ -5,9 +5,19 @@
+import lombok.Setter;
 
 @Data
+@Setter
```
@Setter, @Getter y @EqualsAndHashCode (con el super a false) son redundantes: el @Data los incluye todos.
https://projectlombok.org/features/Data
(NOTA: En entidades el @Data no se utiliza pk también incluye el @ToString que puede dar problemas con hibernate).

amarvila commented on project-x/src/main/java/com/everis/d4i/project_x/service/impl/CustomerServiceImpl.java:
```diff
@@ -40,23 +45,27 @@
 
     @Override
     public CustomerRest createCustomer(final CustomerRest customerRest) throws SalesException {
```
El servicio debería trabajar únicamente con DTOs para independizarlo de la entrada/salida de datos. De esta forma, si el customerRest cambia por el motivo que sea, el servicio no cambia.
CreateCustomer debería recibir un dto

amarvila replied:
Podrías argumentar que estás utilizando el mapper para pasar de rest a dto, y es cierto, pero hace falta ser coherente en cada nivel: controlador trabaja con rest, servicio con dto, persistencia con entidades. Es al hacer la llamada a otra capa cuando utilizas el mapper, o cuando recibes algo de otra capa distinta

amarvila commented on project-x/src/main/java/com/everis/d4i/project_x/service/impl/CustomerServiceImpl.java:
```diff
@@ -40,23 +45,27 @@
 		.orElseThrow(() -> new SalesNotFoundException(new ErrorDto(ExceptionConstantsUtils.NOT_FOUND_GENERIC)));
-
-	return modelMapper.map(customerRepository.save(updateCustomer(customer, customerRest)), CustomerRest.class);
+	CustomerEntity customerEntity = customerRepository.save(updateCustomer(customer, customerRest));
```
Cuidado, que estás  haciendo cosas raras.. un mapeo de customerRest a customer? En cualquier caso, el save es de una entidad y no me queda claro que es esto XD

amarvila commented on project-x/src/main/java/com/everis/d4i/project_x/service/impl/CustomerServiceImpl.java:
```diff
@@ -40,23 +45,27 @@
-	return modelMapper.map(customerRepository.save(updateCustomer(customer, customerRest)), CustomerRest.class);
+	CustomerEntity customerEntity = customerRepository.save(updateCustomer(customer, customerRest));
+	CustomerDto customerDto = customerEntityMapper.mapToDto(customerEntity);
+	return customerRestMapper.mapToRest(customerDto);
```
Igual que antes... el servicio trabaja con dtos, y devuelve dtos. Es el controlador el que tiene el trabajo de hacer la conversión a rest.

amarvila commented on project-x/src/main/java/com/everis/d4i/project_x/service/model/CustomerDto.java:
```diff
@@ -0,0 +1,28 @@
+import lombok.Setter;
+
+@Data
+@Setter
```
Ditto getters y setters y el resto... como el comentario anterior.

Commit:
Smal fixes

Commit:
Add some fixes & tests

Commit:
Mapper Junit Tests

Commit:
[wip]
crear perfil ejecución local, y perfil de test
corregir paginación y hacer rests serializables
pasar de patch a put en controlador update

Commit:
[wip]
revision tests
